tests/unit/conftest.py

```diff
@@ -314,3 +315,11 @@ def attach_storage(
     patch.container.fs["/proc/mounts"] = "/var/www/html/wp-content/uploads"
     yield
     patch.container.fs["/proc/mounts"] = ""
+
+
+@pytest.fixture(scope="function", name="reset_proxy_env")
+def reset_proxy_env():
+    """Reset proxy related env variables during charm init."""
+    os.environ["JUJU_CHARM_HTTP_PROXY"] = ""
+    os.environ["JUJU_CHARM_HTTPS_PROXY"] = ""
+    os.environ["JUJU_CHARM_NO_PROXY"] = ""
```

Comment:
If you use https://docs.pytest.org/en/7.1.x/how-to/monkeypatch.html then pytest will automatically reverse any changes made after the test completes

Reply:
Thanks, I updated the tests to use monkeypatch.setenv